tests/lib/rules/default-props-match-prop-types.js

```diff
@@ -683,6 +683,26 @@ ruleTester.run('default-props-match-prop-types', rule, {
       parser: 'babel-eslint'
     },
     {
+      code: `
+        type PropsA = { foo?: string };
+        type PropsB = { bar?: string, fooBar: string };
+        type Props = PropsA & PropsB;
+
+        class Bar extends React.Component {
+          props: Props;
+          static defaultProps = {
+            foo: "foo",
+            bar: "bar",
```

Comment:
Can you add a test case where a defaultProp is omitted for a `?` prop type, and there's no error? Alternatively, you could just remove the defaultProp for "bar" here.